python/lancedb/embeddings/fine_tuner/dataset.py

```diff
@@ -0,0 +1,98 @@
+import json
+import uuid
+import re
+from tqdm import tqdm
+from pydantic import BaseModel
+from typing import Dict, List, Tuple
+
+from .llm import Openai, BaseLLM
+
+DEFAULT_PROMPT_TMPL = """\
+Context information is below.
+
+---------------------
+{context_str}
+---------------------
+
+Given the context information and not prior knowledge.
+generate only questions based on the below query.
+
+You are a Teacher/ Professor. Your task is to setup \
+{num_questions_per_chunk} questions for an upcoming \
+quiz/examination. The questions should be diverse in nature \
+across the document. Restrict the questions to the \
+context information provided."
+"""
+
+
+class QADataset(BaseModel):
+    """Embedding QA Finetuning Dataset.
+
+    Args:
+        queries (Dict[str, str]): Dict id -> query.
+        corpus (Dict[str, str]): Dict id -> string.
+        relevant_docs (Dict[str, List[str]]): Dict query id -> list of doc ids.
+
+    """
+
+    queries: Dict[str, str]  # dict id -> query
+    corpus: Dict[str, str]  # dict id -> string
+    relevant_docs: Dict[str, List[str]]  # query id -> list of doc ids
+    mode: str = "text"
```

Comment:
Here `mode` is similar to llama-index QADataset which I think is meant to be used when supporting multi-modal/vision embedding model fine-tuning

Reply:
yes..haven't thought deeply about it but I think they should also be multi-modal..where string would represent the path of images